### PR TITLE
Fix MySiteViewController memory leak

### PR DESCRIPTION
**Pull request by salimbraksa**
Fixes #

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift [hunk 1]
```diff
@@ -121,11 +121,17 @@ final class MySiteViewController: UIViewController, UIScrollViewDelegate, NoSite
...
+        }
+        let launchLoginForSelfHostedSite: () -> Void = { [weak self] in
+            self?.launchLoginForSelfHostedSite()
+        }
         let configuration = AddNewSiteConfiguration(
             canCreateWPComSite: viewModel.defaultAccount != nil,
             canAddSelfHostedSite: AppConfiguration.showAddSelfHostedSiteButton,
-            launchSiteCreation: self.launchSiteCreationFromNoSites,
-            launchLoginForSelfHostedSite: self.launchLoginForSelfHostedSite
+            launchSiteCreation: launchSiteCreation,
```
Comment (staskus) [inline WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift:133]: This is classic. Did it solve memory leaks or are there more issues?

Comment (salimbraksa) [reply WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift:133]: The issue is resolved, but there's a risk of it breaking if the view controller leaks again. I'm exploring ways to address the symptoms of this bug if `MySiteViewController` leaks once more.

Comment (staskus) [reply WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift:133]: Got it. Also, it's interesting if it's possible to write test(s) to detect such leaks happening. 🤔 

I tried adding [these tests to WordPressAuthenticator](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/trunk/WordPressAuthenticatorTests/MemoryManagementTests.swift) but they only catch leaks that happen when a view controller is initialized and loaded, and not something that happens after custom actions are made. 

For example, this `noSitesViewController` is `lazy var` which is only initialized when we want to present empty state, therefore such simple unit tests wouldn't catch it.

**Comment by mokagio** [comment]
Moving this to 24.0 as I'll run the 23.9 code freeze and this has not been approved yet.

**Comment by mokagio** [comment]
I'm going to bump this to the next release because we'll be code freezing [24.0](https://github.com/wordpress-mobile/WordPress-iOS/milestone/265) today and this is still a draft.

**Pull request closed without merging**